#!/usr/bin/env python
PACKAGE = "rc_hand_eye_calibration_client"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

axis_enum = gen.enum([gen.const("off", int_t, -1, "4DOF off"),
		      gen.const("X", int_t, 0, "x axis"),
		      gen.const("Y", int_t, 1, "y axis"),
                      gen.const("Z", int_t, 2, "z axis")],
		      "Enum to set the tcp rotation axis" )

# name, type,level, descriptiion, default, min, max
gen.add("grid_width",      double_t, 0, "The width of the calibration pattern in meters", 0.0, 0.0, 10.0)
gen.add("grid_height",     double_t, 0, "The height of the calibration pattern in meters", 0.0, 0.0, 10.0)
gen.add("robot_mounted",   bool_t,   0, "Whether the camera is mounted on the robot",  True)
gen.add("tcp_rotation_axis", int_t, 0, "TCP rotation axis for 4 DOF robot calibration (-1 for general robot)", -1, -1, 2, edit_method=axis_enum)
gen.add("tcp_offset",   double_t,   0, "Offset from the TCP for 4 DOF robot calibration", 0.0, -10.0, 10.0)

exit(gen.generate(PACKAGE, "rc_hand_eye_calibration_client", "hand_eye_calibration"))
